x/did/keeper/msg_server.go

```diff
@@ -149,3 +153,45 @@ func VerifyAllSignersHaveAtLeastOneValidSignature(k *Keeper, ctx *sdk.Context, i
 
 	return nil
 }
+
+func (k MsgServer) Burn(goCtx context.Context, msg *types.MsgBurn) (*types.MsgBurnResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(goCtx)
+	accountI := k.Keeper.accountKeeper.GetAccount(sdkCtx, sdk.AccAddress(msg.FromAddress))
+	_, ok := accountI.(authtypes.ModuleAccountI)
+	if ok {
+		return nil, types.ErrBurnFromModuleAccount
+	}
+
+	bondDenom := k.stakingKeeper.BondDenom(sdkCtx)
+	denoms := msg.Amount.Denoms()
+	if len(denoms) != 0 {
+		err := ValidateDenom(denoms[0], bondDenom)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Similarly, there might exist more than one records, so all _must_ be validated accordingly rather than just the first list element.